File: backend/lib/discounts.py
"""Cart discount logic — ported from prediosChile."""

DISCOUNT_TIERS = [
    (20, 0.15),  # 15% off for 20+ items
    (10, 0.10),  # 10% off for 10+ items
    (3, 0.05),   # 5% off for 3+ items
]


def calculate_cart_total(items: list[dict]) -> dict:
    """Calculate cart total with volume discount.

    Each item must have a 'precio' key.
    Returns: {subtotal, discount, discount_amount, total}
    """
    subtotal = sum(item.get("precio", 0) for item in items)
    count = len(items)

    discount = None
    for min_items, pct in DISCOUNT_TIERS:
        if count >= min_items:
            discount = {"min": min_items, "pct": pct}
            break

    discount_amount = int(subtotal * discount["pct"]) if discount else 0
    total = subtotal - discount_amount

    return {
        "subtotal": subtotal,
        "discount": discount,
        "discount_amount": discount_amount,
        "total": total,
    }

File: frontend/src/utils/discounts.js
export const DISCOUNT_TIERS = [
  { min: 20, pct: 0.15, label: '15% dcto (20+ comunas)' },
  { min: 10, pct: 0.10, label: '10% dcto (10+ comunas)' },
  { min: 3, pct: 0.05, label: '5% dcto (3+ comunas)' },
]

export function calculateCartTotal(items) {
  const subtotal = items.reduce((sum, item) => sum + (item.precio || 0), 0)
  const count = items.length

  let discount = null
  for (const tier of DISCOUNT_TIERS) {
    if (count >= tier.min) {
      discount = tier
      break
    }
  }

  const discountAmount = discount ? Math.floor(subtotal * discount.pct) : 0
  const total = subtotal - discountAmount

  return { subtotal, discount, discountAmount, total }
}
